proxy/src/http.rs

```diff
@@ -21,6 +24,23 @@ pub fn new_client() -> ClientWithMiddleware {
         .build()
 }
 
+pub fn new_client_with_timeout_and_retries(
+    default_timout: Duration,
+    total_retries: u32,
+) -> ClientWithMiddleware {
+    let timeout_client = reqwest::ClientBuilder::new()
+        .timeout(default_timout)
+        .build()
+        .expect("Failed to create http client with timeout");
+
+    let retry_policy = ExponentialBackoff::builder().build_with_max_retries(total_retries);
+
```

Comment:
I just reviewed the `reqwest_retry` crate's docs.
IIUC this will set a fixed number of retries while leaving default inter-retry inerval bounds and backoff exponent.

Default max inter-retry interval bound is 30s.

We don't cap the max time spent on retries.
Is that intentional?

Should we use https://docs.rs/reqwest-retry/latest/reqwest_retry/policies/struct.ExponentialBackoffBuilder.html#method.build_with_total_retry_duration instead?

I think the latter makes more sense because we're running as part of a loop with a defined period.

What's more important? Getting metrics out with more retries, thereby blowing past the loop period, or sticking to the loop period?

Reply:
(If build_with_max_retries is intentional, maybe just add a comment?)

Reply:
lets stick to loop period as this sparked all of this change